subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/connection/LazyConsumerActionExecutor.java

```diff
@@ -49,11 +52,14 @@
     private final ConnectionParameters connectionParameters;
     private BuildCancellationToken cancellationToken;
 
+    private final ExecutorService executorService;
+
     public LazyConsumerActionExecutor(Distribution distribution, ToolingImplementationLoader implementationLoader, LoggingProvider loggingProvider, ConnectionParameters connectionParameters) {
         this.distribution = distribution;
         this.implementationLoader = implementationLoader;
         this.loggingProvider = loggingProvider;
         this.connectionParameters = connectionParameters;
+        this.executorService = Executors.newSingleThreadExecutor();
```

Comment:
when does this executor get shutdown?

Reply:
You are rigth, it was missing. I've fixed that by inlining the excutor and immediately shutting it down.
Maybe we should add the executor as a service shared between `LazyConsumerActionExecutor` instances, but it might be an overkill. WDYT?